src/pub_verif_token.ts

```diff
@@ -247,10 +388,73 @@ export class Client {
             tokRes.blindSig,
             this.finData.inv,
         );
-        const token = new Token(BLIND_RSA, this.finData.authInput, authenticator);
+        const token = new Token(this.tokenType, this.finData.authInput, authenticator);
 
         this.finData = undefined;
 
         return token;
     }
 }
+
+export class Client extends PubliclyVerifiableClient {
+    constructor(mode: BlindRSAMode | PartiallyBlindRSAMode) {
+        super(mode);
+    }
+
+    async createTokenRequest(
+        tokChl: TokenChallenge,
+        issuerPublicKey: Uint8Array,
+    ): Promise<TokenRequest> {
+        return super._createTokenRequest(tokChl, issuerPublicKey);
+    }
+}
+
+export class ClientWithMetadata extends PubliclyVerifiableClient {
+    constructor(mode: BlindRSAMode | PartiallyBlindRSAMode, extensions: Extensions) {
+        super(mode, extensions);
+    }
+
+    async createTokenRequest(
+        tokChl: TokenChallenge,
+        issuerPublicKey: Uint8Array,
+    ): Promise<ExtendedTokenRequest> {
+        const tokenRequest = await super._createTokenRequest(tokChl, issuerPublicKey);
+        return new ExtendedTokenRequest(tokenRequest, this.extensions!);
+    }
+}
+
+abstract class PubliclyVerifiableOrigin {
+    private suite: Pick<BlindRSA, 'verify'>;
+
+    constructor(
+        public readonly mode: BlindRSAMode | PartiallyBlindRSAMode,
```

Comment:
there is some `originInfo` defined in the examples than need to be set here.

Reply:
updated